lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -969,15 +970,48 @@ impl LocalShard {
         vector_size * info.points_count
     }
 
-    pub async fn local_shard_info(&self) -> CollectionInfoInternal {
+    pub async fn local_shard_status(&self) -> (ShardStatus, OptimizersStatus) {
+        let mut status = ShardStatus::Green;
+        let mut optimizer_status = OptimizersStatus::Ok;
+
+        {
+            let segments = self.segments().read();
+            for (_, segment) in segments.iter() {
+                let segment_info = segment.get().read().info();
+                if segment_info.segment_type == SegmentType::Special {
+                    status = ShardStatus::Yellow;
+                }
+
+                if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
+                    status = ShardStatus::Red;
+                }
+
+                if let Some(error) = &segments.optimizer_errors {
+                    optimizer_status = OptimizersStatus::Error(error.to_string());
+                }
+            }
+        }
+
+        // If still green while optimization conditions are triggered, mark as grey
+        if status == ShardStatus::Green
+            && optimizer_status == OptimizersStatus::Ok
+            && self.update_handler.lock().await.has_pending_optimizations()
+        {
+            status = ShardStatus::Grey;
+            optimizer_status =
+                OptimizersStatus::Error("optimizations pending, awaiting update operation".into());
```

Comment:
This was only here temporary while we couldn't use the grey status. I suggest to remove this now.

```suggestion
```

Reply:
done